image-rs/src/image.rs

```diff
@@ -93,23 +94,16 @@ pub struct ImageClient {
 
     /// The config
     pub(crate) config: ImageConfig,
+
+    /// The image layer store
+    pub(crate) layer_store: Arc<LayerStore>,
```

Comment:
I don't think the we need the Arc around the whole layer store, it should be sufficient to only wrap the `AtomicUsize`. Please see the attached patch. We can pass around the layer store via ref

[arc_for_atomicusize_only.patch](https://github.com/user-attachments/files/18797207/arc_for_atomicusize_only.patch)

Reply:
I am thinking whether we need Arc. AtomicUsize should be thread safe.

Reply:
Now I remember why I changed from lifetime to Arc.  There is validation [error](https://github.com/confidential-containers/guest-components/actions/runs/13330930001/job/37234635058?pr=902) which I do not quite understand.

Any suggestions how to handle it?

Reply:
While types like `AtomicUsize` provide thread-safe operations, sharing them across concurrently running tasks (like our image pulling algo) requires shared ownership semantics. `Arc` ensures that the data remains valid until all tasks are finished using it.